### PR TITLE
Support output-type for TOC

### PR DESCRIPTION
**Pull request by AngryBerryMS**
* no template when output-type is "json"
* add a schema named "toc" to the output json

### CODE REVIEW

File: src/docfx/build/toc/TocBuilder.cs [hunk 1]
```diff
@@ -85,7 +85,15 @@ public void Build(ErrorBuilder errors, FilePath file)
                     var output = _templateEngine.RunJavaScript("toc.json.js", JsonUtility.ToJObject(model));
                     _output.WriteJson(Path.ChangeExtension(outputPath, ".json"), output);
                 }
-                else
+
+                if (_config.OutputType == OutputType.Json)
+                {
+                    var outputModel = JsonUtility.ToJObject(model);
+                    outputModel["schema"] = "toc";
```
Comment (yufeih) [inline src/docfx/build/toc/TocBuilder.cs:92]: Add a get only property to `TocModel` and init the value to `toc`, to line 79/85/98 all pick the `schema` property.

File: docs/specs/toc/toc.yml [hunk 2]
```diff
@@ -1522,3 +1522,16 @@ outputs:
         "a.md": [{ "source": "TOC.yml", "type": "metadata" }]
       }
     }
+---
```
Comment (yufeih) [inline docs/specs/toc/toc.yml:1525]: Also add a test case with `output-type=pagejson`

Comment (AngryBerryMS) [reply docs/specs/toc/toc.yml:1525]: I will change the default output-type as "pageJson" and then no need to add new spec tests for it.

File: docs/specs/toc/toc.yml [hunk 3]
```diff
@@ -1522,3 +1522,16 @@ outputs:
...
+inputs:
+  docfx.yml: |
+    outputType: Json
+  TOC.yml: |
+    - name: a
+      href: a.md
+  a.md:
+outputs:
+  a.json:
+  toc.json: |
```
Comment (yufeih) [inline docs/specs/toc/toc.yml:1536]: Also need a test in `static.yml` to cover `outputType=html` + `toc.json.js`

File: docs/specs/toc/toc.yml [hunk 4]
```diff
@@ -1522,3 +1522,16 @@ outputs:
...
+---
+# TOC with output-type="json"
+inputs:
+  docfx.yml: |
+    outputType: Json
+  TOC.yml: |
+    - name: a
+      href: a.md
+  a.md:
+outputs:
```
[the remark this answers is not in the document]
Comment (AngryBerryMS) [reply docs/specs/toc/toc.yml:1546]: Done.

File: docs/specs/toc/toc.yml (hunk 3 above)
Comment (AngryBerryMS) [reply docs/specs/toc/toc.yml:1548]: I add a mustache template file with a js file to generate `.html` and `.json` outputs.

File: src/docfx/build/toc/TocBuilder.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/build/toc/TocBuilder.cs]: Done.

File: src/docfx/build/toc/TocModel.cs [hunk 5]
```diff
@@ -14,6 +14,8 @@ internal class TocModel
         [JsonProperty("_path")]
         public string Path { get; }
 
+        public string Schema { get; init; } = "toc";
```
Comment (yufeih) [inline src/docfx/build/toc/TocModel.cs:17]:
```suggestion
        public string Schema { get; } = "toc";
```

**Review by yufeih — APPROVED** [review]

**Pull request merged**